pandas/_libs/ops.pyx

```diff
@@ -160,17 +160,19 @@ def vec_compare(ndarray[object] left, ndarray[object] right, object op) -> ndarr
         for i in range(n):
             x = left[i]
             y = right[i]
-
-            if checknull(x) or checknull(y):
+            if x is None and y is None:
```

Comment:
this is superfluous as checknull already checks None

Reply:
Yep, this was incorrect. Changed.

Reply:
what was incorrect?

rule of thumb: let the question-asker mark the conversation as resolved